### PR TITLE
Feature: Validations and Model Specs

### PR DESCRIPTION
**Pull request by Akorede3133**
# In the PR, the following were affected:

## Added the following validations:
* For the User model:
    * Name must not be blank.
    * PostsCounter must be an integer greater than or equal to zero.
* For the Post model:
    * Title must not be blank.
    * Title must not exceed 250 characters.
    * CommentsCounter must be an integer greater than or equal to zero.
    * LikesCounter must be an integer greater than or equal to zero.
* Added unit specs for all of your models' methods and validations.

### CODE REVIEW

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Akorede, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job writing the validations.

## Required Changes ♻️
- [ ] Please make sure to write unit tests for the comment and like model as well and put these in separate spec files.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/model/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  it 'is not valid when posts_counter is not an integer' do
+    subject.posts_counter = 'a'
+    expect(subject).not_to be_valid
+  end
+
+  it 'is not valid when posts_counter is less than zero' do
+    subject.posts_counter = -1
+    expect(subject).not_to be_valid
```
Comment (ghost) [inline spec/model/user_spec.rb:24]: - [ ] Good work adding tests to your models. Kindly also remember that any `custom methods` in your models should be tested. In your `user model`, kindly test the `most_recent_posts` method

File: spec/model/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  it 'is not valid when comments_counter is not an integer' do
+    subject.comments_counter = 'a'
+    expect(subject).not_to be_valid
+  end
+
+  it 'is not valid when comments_counter is less than zero' do
+    subject.comments_counter = -10
+    expect(subject).not_to be_valid
+  end
```
Comment (ghost) [inline spec/model/post_spec.rb]: - [ ] Kindly test the `update_posts_counter` and  `most_recent_comments`  methods in your `Post` model

**Review by ghost — CHANGES_REQUESTED** [review]
Hi @Akorede3133 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

✅ User model has validations
✅ Post model has validations
✅ Model specs has Validation tests 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @Akorede3133,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**